plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/LocalAppEngineServerLaunchConfigurationDelegate.java

```diff
@@ -83,42 +84,25 @@ public void launch(ILaunchConfiguration configuration, String mode, ILaunch laun
     }
   }
 
-  private void runDebugTarget(LocalAppEngineServerDelegate serverDelegate, String projectName, int port)
+  private void setupDebugTarget(ILaunch launch, ILaunchConfiguration configuration, LocalAppEngineServerDelegate serverDelegate, String projectName, int port, IProgressMonitor monitor)
       throws CoreException {
-    ILaunchConfigurationWorkingCopy remoteDebugLaunchConfig = createRemoteDebugLaunchConfiguration(serverDelegate,
-        projectName,
-        Integer.toString(port));
-    remoteDebugLaunchConfig.launch(ILaunchManager.DEBUG_MODE, null);
-  }
-
-  private ILaunchConfigurationWorkingCopy createRemoteDebugLaunchConfiguration(LocalAppEngineServerDelegate serverDelegate,
-      String projectName, String port) throws CoreException {
-    ILaunchManager manager = DebugPlugin.getDefault().getLaunchManager();
-    ILaunchConfigurationType type = manager.getLaunchConfigurationType(IJavaLaunchConfigurationConstants.ID_REMOTE_JAVA_APPLICATION);
-
-    ILaunchConfigurationWorkingCopy remoteDebugConfig = type.newInstance(null,
-        "Remote debugger for "
-            + projectName
-            + " ("
-            + port
-            + ")");
-
-    // Set project
-    remoteDebugConfig.setAttribute(IJavaLaunchConfigurationConstants.ATTR_PROJECT_NAME,
-        projectName);
 
     // Set JVM debugger connection parameters
+    int timeout = JavaRuntime.getPreferences().getInt(JavaRuntime.PREF_CONNECT_TIMEOUT);
     Map<String, String> connectionParameters = new HashMap<>();
     connectionParameters.put("hostname", DEBUGGER_HOST);
-    connectionParameters.put("port", port);
-    remoteDebugConfig.setAttribute(IJavaLaunchConfigurationConstants.ATTR_CONNECT_MAP,
-        connectionParameters);
-    remoteDebugConfig.setAttribute(IJavaLaunchConfigurationConstants.ATTR_VM_CONNECTOR,
-        "org.eclipse.jdt.launching.socketListenConnector");
-
-    return remoteDebugConfig;
+    connectionParameters.put("port", Integer.toString(port));
+    connectionParameters.put("timeout", Integer.toString(timeout));
+    
+    IVMConnector connector = JavaRuntime.getVMConnector("org.eclipse.jdt.launching.socketListenConnector");
+    if(connector == null) {
+      abort("Cannot find Socket Listening connector", null, 0);
+    } else {
+      connector.connect(connectionParameters, monitor, launch);
```

Comment:
probably irrelevant, but you could move most of the code before the if block into the else statement. I.e. it doesn't matter if the connector isn't found.